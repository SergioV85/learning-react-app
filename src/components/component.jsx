import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}