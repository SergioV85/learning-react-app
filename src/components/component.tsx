import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}