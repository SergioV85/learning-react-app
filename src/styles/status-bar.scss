@import './variables';

.c-status-bar {
  align-items: center;
  display: flex;
  justify-content: space-between;

  &__movies-quantity {
    color: $primary-dark;
    font-family: $main-font;
    font-size: 0.9rem;
    font-weight: 700;
  }
}