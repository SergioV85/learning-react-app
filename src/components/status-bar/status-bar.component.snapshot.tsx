import { shallow } from 'enzyme';
import * as React from 'react';
import { StatusBarPanel } from './status-bar.component';

describe('Status Bar Panel Component Snapshot', () => {
  const mockedChangeSorting = jest.fn();

  it('should renders correctly', () => {
    const statusBarPanelSnapshot = shallow(
      <StatusBarPanel foundMovies={2} currentSorting='genre' onChangeSorting={mockedChangeSorting} />,
    );
    expect(statusBarPanelSnapshot)
      .toMatchSnapshot();
  });
});
