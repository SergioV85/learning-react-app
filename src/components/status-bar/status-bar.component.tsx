import * as React from 'react';
import { ToggleSwitcher } from './../../shared-components/toggle-switcher/toggle-switcher.component';

interface IStatusBarProps {
  foundMovies: number;
  currentSortingOrder: string;
  currentSortingType: string;
  onChangeSortingOrder(): void;
  onChangeSortingType(): void;
}

export class StatusBarPanel extends React.Component<IStatusBarProps, {}> {
  private sortingOrderButtons = [
    {
      icon: 'icon-arrow-up',
      value: 'asc',
    },
    {
      icon: 'icon-arrow-down',
      value: 'desc',
    },
  ];
  private sortingTypeButtons = [
    {
      label: 'Release date',
      value: 'release_date',
    },
    {
      label: 'Rating',
      value: 'rating',
    },
  ];

  constructor(props: IStatusBarProps) {
    super(props);
  }

  public render() {
    if (this.props.foundMovies > 0) {
      return <div className='c-status-bar p-2 container'>
        <span className='c-status-bar__movies-quantity'>{this.props.foundMovies} movies found</span>
        <ToggleSwitcher
            cssClass='c-status-bar__sorting-type'
            title='Sort by'
            buttonViewClass='btn-link'
            buttons={this.sortingTypeButtons}
            currentValue={this.props.currentSortingType}
            name='sort-type'
            type='radio'
            onToggleChange={this.props.onChangeSortingType}
        />
        <ToggleSwitcher
            cssClass='c-status-bar__sorting-order'
            title='Sort order'
            buttonViewClass=''
            buttons={this.sortingOrderButtons}
            currentValue={this.props.currentSortingOrder}
            name='sort-order'
            type='radio'
            onToggleChange={this.props.onChangeSortingOrder}
        />
      </div>;
    }
    return null;
  }
}
