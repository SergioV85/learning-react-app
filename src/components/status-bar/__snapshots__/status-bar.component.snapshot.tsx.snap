// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Bar Panel Component Snapshot should renders correctly 1`] = `
<div
  className="c-status-bar p-2 container"
>
  <span
    className="c-status-bar__movies-quantity"
  >
    2
     movies found
  </span>
  <ToggleSwitcher
    buttonViewClass="btn-link"
    buttons={
      Array [
        Object {
          "label": "Release date",
          "value": "release_date",
        },
        Object {
          "label": "Rating",
          "value": "rating",
        },
      ]
    }
    cssClass="c-status-bar__sorting-type"
    currentValue="rating"
    name="sort-type"
    onToggleChange={[MockFunction]}
    title="Sort by"
    type="radio"
  />
  <ToggleSwitcher
    buttonViewClass=""
    buttons={
      Array [
        Object {
          "icon": "icon-arrow-up",
          "value": "asc",
        },
        Object {
          "icon": "icon-arrow-down",
          "value": "desc",
        },
      ]
    }
    cssClass="c-status-bar__sorting-order"
    currentValue="asc"
    name="sort-order"
    onToggleChange={[MockFunction]}
    title="Sort order"
    type="radio"
  />
</div>
`;
